### PR TITLE
Major update to Hal Components

### PR DESCRIPTION
**Pull request by GomezIvann**
### Summary of changes

- **Updated libraries:**
  - `@dxc-technology/halstack-react` updated to 8.0.0.
  - `@dxc-technology/halstack-react-client` updated to 1.3.3.
- **Clean up the library's package.json.**
- **Updated specs for the HalTable to match the current implementation and design of the DxcResultsetTable.**
  - Updated the error font colour to match the DS error patterns and colour standards.
- **New Typescript implementation.**
  - Now the Hal Components export types, which make them more secure when programming (type-checking, suggestions, comments, etc).
  - Updated build and configuration files to correctly export the typing.
  - /app example project has also been updated to use Typescript.
- **Updated README to match the new types and some typo fixes.** 
- **Majors:**
  - `HalAutocomplete` prop `url` has been renamed to `collectionUrl` to match the rest of the hal components.

### CODE REVIEW

File: lib/package.json [hunk 1]
```diff
@@ -9,22 +9,21 @@
     "url": "https://dxc.com"
   },
   "main": "./dist/main.js",
+  "types": "./dist/main.d.ts",
   "license": "Apache-2.0",
   "peerDependencies": {
     "react": "^18.x",
     "react-dom": "^18.x",
-    "@dxc-technology/halstack-react": "6.x || 0.0.0-x"
+    "@dxc-technology/halstack-react": "8.0.0"
```
Comment (jsuarezgonz) [inline lib/package.json:17]: We should set the version to "^8.0.0" just for compatibility reasons.

**Comment by kukoc** [comment]
The colors or the error fonts are different for the Table and the autocomplete. is there any reason for that?

**Review by kukoc — APPROVED** [review]

File: lib/package.json (hunk 1 above)
Comment (GomezIvann) [reply lib/package.json]: Done!

**Comment by GomezIvann** [comment]
> The colors or the error fonts are different for the Table and the autocomplete. is there any reason for that?

I decided to go for the error font colour because the HalTable had that text in a red colour previously, so to change the design as less as possible, I only updated that red colour to the new value in the CDK. In the case of the Text Input, that was a design decision made by Alex.

**Review by kukoc — APPROVED** [review]

**Pull request merged**